arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f446xx

```diff
@@ -0,0 +1,31 @@
+# Kconfig - ST STM32F446XX MCU configuration options
+#
+# Copyright (c) 2017 Pushpal Sidhu
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if SOC_STM32F446XE
+
+config SOC
+	string
+	default stm32f446xe
```

Comment:
Should be stm32f446xx:
CONFIG_SOC is used in ext/hal/st/stm32cube/Makefile to generate a SoC specific compile flag.
This compile flag exposes SoC specific features in stm32f4xx SDK and represents "granularity" of the SoC in the family.
In case of stm32f446xx SoCs there is no difference from SW point of view among the variants. 
Hence here you should use stm32f446xx, which will be generated in STM32F446xx define, which is used in stm32f4xx cube hal to define specific features for this SoC and its derivates

Reply:
One question: The XE and the XC have different flash sizes. (256 vs 512). I see that in mem.h that the CONFIG_SOC is used to determine what to set the size to. How should we overcome that?

Reply:
Flash Size information (along with RAM size) will be guessed thanks to CONFIG_SOC_STM32F..... flags generated in Kconfig.soc file and used in dts/arm/st/mem.h.
This will be used by "OS side" of Zephyr

Now, regarding pure STM32Cube HAL point of view, Flash size has no impact on STM32F446xx variants and this is why the information is not collected and generic "xx" is used.
HIH